src/autoscan_inotify.cc

```diff
@@ -325,7 +325,7 @@ void AutoscanInotify::monitorNonexisting(const fs::path& path, const std::shared
 void AutoscanInotify::recheckNonexistingMonitor(int curWd, std::vector<std::string> pathAr, const std::shared_ptr<AutoscanDirectory>& adir)
 {
     bool first = true;
-    for (int i = pathAr.size(); i >= 0; i--) {
+    for (int i = int(pathAr.size()); i >= 0; i--) {
```

Comment:
try to change the type to size_t

Reply:
That would break the if condition as size_t cannot go negative.

Reply:
It works on my side if you change the type of `j` in line 333 also to `size_t`

Reply:
That's not the issue. i >= 0; is always true if i is unsigned, causing an infinite loop (potentially). clang also warns about this.